memory.cpp

```diff
@@ -253,36 +253,36 @@
 
             void InitHooks()
             {
-                g_origMalloc = __malloc_hook;
-                g_origFree = __free_hook;
-                __malloc_hook = MallocHook;
-                __free_hook = FreeHook;
+                // g_origMalloc = __malloc_hook;
+                // g_origFree = __free_hook;
+                // __malloc_hook = MallocHook;
+                // __free_hook = FreeHook;
             }
             void (* volatile __malloc_initialize_hook)() = InitHooks;
 
-            void* MallocHook(size_t size, const void * context)
-            {
-                if (pthread_getspecific(tls_key) || HasSEAInStack())
-                    return g_origMalloc(size, context);
-                CRecursionScope scope;
-
-                __itt_heap_allocate_begin(g_heap, size, 0);
-                void* res = g_origMalloc(size, context);
-                __itt_heap_allocate_end(g_heap, &res, size, 0);
-
-                return res;
-            }
-
-            void FreeHook(void* ptr, const void* context)
-            {
-                if (pthread_getspecific(tls_key) || HasSEAInStack())
-                    return g_origFree(ptr, context);
-                CRecursionScope scope;
-
-                __itt_heap_free_begin(g_heap, ptr);
-                g_origFree(ptr, context);
-                __itt_heap_free_end(g_heap, ptr);
-            }
+            // void* MallocHook(size_t size, const void * context)
+            // {
+            //     if (pthread_getspecific(tls_key) || HasSEAInStack())
+            //         return g_origMalloc(size, context);
+            //     CRecursionScope scope;
+            //
+            //     __itt_heap_allocate_begin(g_heap, size, 0);
+            //     void* res = g_origMalloc(size, context);
+            //     __itt_heap_allocate_end(g_heap, &res, size, 0);
+            //
+            //     return res;
+            // }
+            //
+            // void FreeHook(void* ptr, const void* context)
+            // {
+            //     if (pthread_getspecific(tls_key) || HasSEAInStack())
+            //         return g_origFree(ptr, context);
+            //     CRecursionScope scope;
+            //
+            //     __itt_heap_free_begin(g_heap, ptr);
+            //     g_origFree(ptr, context);
+            //     __itt_heap_free_end(g_heap, ptr);
+            // }
```

Comment:
Musl libc does not support malloc hooks; this needs conditional compilation.

Reply:
yes, please just add the conditional compilation for Musl around these blocks, this will do.
Best if there is a way to check the support of malloc hooks and the backtrace.

Reply:
@raphaelcohn, if I understand all of this correctly, we need to add an option to `ittnotify/CMakeLists.txt` so that we can turn on MUSL compilation when needed. So would something like the following work for you?

```
option(USE_MUSL_LIBC "Use musl's libc implementation during compilation" OFF)
IF(USE_MUSL_LIBC)
    SET(CC "musl-gcc")  # I assume this and the `-static -Os` flags are necessary due to https://stackoverflow.com/a/41873196
    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -static -Os -DUSE_MUSL_LIBC")  # by defining USE_MUSL_LIBC we can conditionally compile all that commented out code
ENDIF()
```

I haven't tested the above but it apparently should run with something like `cmake -DUSE_MUSL_LIBC=ON ...`. What do you think?

Reply:
(I'm under the impression that you only really need to build `ittnotify` but this may not be the case and it could be other `CMakeLists.txt` files--or the top-level one--that need to have something like the above; the same idea should apply).

Reply:
yeah, let's not split the compilation of ITT from ISEA